homeassistant/components/sonos/media_player.py

```diff
@@ -777,6 +778,27 @@ async def async_join_players(self, group_members):
             await self.hass.async_add_executor_job(self.speaker.join, speakers)
 
     async def async_unjoin_player(self):
-        """Remove this player from any group."""
-        async with self.hass.data[DATA_SONOS].topology_condition:
-            await self.hass.async_add_executor_job(self.speaker.unjoin)
+        """Remove this player from any group.
+
+        Coalesces all calls within 0.5s to allow use of SonosSpeaker.unjoin_multi()
+        which optimizes the order in which speakers are removed from their groups.
+        Removing coordinators last better preserves playqueues on the speakers.
+        """
+        sonos_data = self.hass.data[DATA_SONOS]
```

Comment:
This maintains data for multiple households ?

Reply:
Yes, but only a few of the data structures need to be separated between households (e.g., S1/S2 systems in the same network). Speakers can't be grouped across households, for example.

I don't have a split setup myself, so this is a defensive measure for the few setups that might encounter this.